build: fix warning of invalid copy

https://github.com/4paradigm/OpenMLDB/blob/607f32b9f20d6610dddee35080ca987225adfb83/src/storage/disk_table.h#L167
```
/Users/ark/Code/OpenMLDB2/src/storage/disk_table.h:167:29: warning: loop variable 'index' of type 'const
      std::__1::shared_ptr<openmldb::storage::IndexDef>' creates a copy from type 'const
      std::__1::shared_ptr<openmldb::storage::IndexDef>' [-Wrange-loop-analysis]
            for (const auto index : indexs) {
```

--- a/src/storage/disk_table.h
+++ b/src/storage/disk_table.h
@@ -164,7 +164,7 @@ class AbsoluteTTLCompactionFilter : public rocksdb::CompactionFilter {
             uint32_t ts_idx = *((uint32_t*)(key.data() + key.size() - TS_LEN -  // NOLINT
                                           TS_POS_LEN));
             bool has_found = false;
-            for (const auto index : indexs) {
+            for (const auto& index : indexs) {
                 auto ts_col = index->GetTsColumn();
                 if (!ts_col) {
                     return false;

